hack/update-image-vars.sh

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+
+set -xeuo pipefail
+
+SFI_REF="${1:-master}"
+SFI_GIT_URI=${SFI_GIT_URI:-https://github.com/openshift/splunk-forwarder-images}
+SFI_DIR=build/_sfi
+
+usage() {
+    echo "Usage: $0 FORWARDER_IMAGE_GIT_COMMIT-ISH" >&2
+    exit -1
+}
+
+clone_or_fetch_sfi_repo() {
+    if [ -d ${SFI_DIR} ]; then
+        git --git-dir ${SFI_DIR}/.git fetch
+    else
+        git clone ${SFI_GIT_URI} ${SFI_DIR}
+    fi
+    (cd ${SFI_DIR} && git checkout --force --quiet $1)
+}
+
+[[ $# -gt 1 ]] || [[ "$*" = "-h" ]] || [[ "$*" = "--help" ]] && usage
+
+# Set SED variable
+if LANG=C sed --help 2>&1 | grep -q GNU; then
+  SED="sed"
+elif command -v gsed &>/dev/null; then
+  SED="gsed"
+else
+  echo "Failed to find GNU sed as sed or gsed. If you are on Mac: brew install gnu-sed." >&2
+  exit 1
+fi
+
+clone_or_fetch_sfi_repo $SFI_REF
+
+forwarder_version=$(<${SFI_DIR}/.splunk-version)
+forwarder_hash=$(<${SFI_DIR}/.splunk-version-hash)
+commit_hash=$(git --git-dir ${SFI_DIR}/.git rev-parse --short=7 HEAD)
+
+${SED?} -i "s,^\(FORWARDER_VERSION\).*=.*$,\1 ?= $forwarder_version," Makefile
+${SED?} -i "s,^\(FORWARDER_HASH\).*=.*$,\1 ?= $forwarder_hash," Makefile
+${SED?} -i "s,^\(SFI_HASH_7\).*=.*$,\1 ?= $commit_hash," Makefile
```

Comment:
To be paranoid, I would match a word boundary after these variable names. Otherwise you run the risk of affecting variables named e.g. `FORWARDER_VERSION_FOO`.
```suggestion
${SED?} -i "s,^\(FORWARDER_VERSION\)\b.*=.*$,\1 ?= $forwarder_version," Makefile
${SED?} -i "s,^\(FORWARDER_HASH\)\b.*=.*$,\1 ?= $forwarder_hash," Makefile
${SED?} -i "s,^\(SFI_HASH_7\)\b.*=.*$,\1 ?= $commit_hash," Makefile
```